pkg/storage/trb_guidance_letter_recommendation.go

```diff
@@ -183,7 +216,16 @@ func (s *Store) UpdateTRBGuidanceLetterRecommendation(ctx context.Context, recom
 			recommendation = :recommendation,
 			links = :links,
 			created_by = :created_by,
-			modified_by = :modified_by
+			modified_by = :modified_by,
+			category = :category,
+			-- update position in letter ONLY when category changes
+			position_in_letter = CASE
+				-- when category changes
+				WHEN category <> :category
+					THEN (SELECT COALESCE(MAX(position_in_letter) + 1, 0) FROM trb_guidance_letter_recommendations WHERE category = :category AND trb_request_id = :trb_request_id)
+				-- when category does not change
+				ELSE position_in_letter
+			END
```

Comment:
You could refactor this into our new sql organization format instead of in-lining it, but up to you.

Reply:
i'd prefer to leave for now and do in a separate PR for that purpose specifically